scripts/force-rebuild-isle.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
+#
+# This script rebuilds all ISLE generated source that is checked in, even if
+# the source has not changed relative to the manifests.
+#
+# This is useful when one is developing the ISLE compiler itself; otherwise,
+# changing the compiler does not automatically change the generated code, even
+# if the `rebuild-isle` feature is specified.
+
+set -e
+
+# Remove the manifests (which contain hashes of ISLE source) to force the build
+# script to regenerate all backends.
+rm -f cranelift/codegen/src/isa/*/lower/isle/generated_code.manifest
```

Comment:
We can future proof this a little for when we have more ISLE than just lowering.

```suggestion
rm -f cranelift/codegen/**/generated_code.manifest
```

Reply:
Hmm, the double-asterisk wildcard seems to work in bash and zsh, but not dash (Ubuntu's default /bin/sh [1]) or 2007-era bash on my macOS machine. So there is no portable shebang line that specifically gets bash on Ubuntu and zsh on macOS.

[1] https://wiki.ubuntu.com/DashAsBinSh

I'll add a comment to the build script instead to note that this should be updated. Thanks for raising the issue in any case!

Reply:
Can we use a

```bash
#!/usr/bin/env bash
```

shebang to reliably get bash across platforms?

Reply:
The issue is that bash on macOS is from 2007 (because GPL 3 I guess) and doesn't have the feature, so we need a different shell on different platforms (`zsh` on modern macOS, `bash` on Linux).

Other environments may not have a new enough shell either (older Linuxes, etc) and I don't really want to introduce a special dependency. Or at least, this isn't really a good enough reason to do so, IMHO.

Reply:
(merging now with your r+ but happy to refine this later if you feel strongly about it; probably would need a Rust script in place of the nonportable wildcard)